### PR TITLE
Added config file to merge command

### PR DESCRIPTION
**Pull request by CalderWhite**
First off: I love pqrs!! Thank you so much for creating it and maintaining it :)

The code in this PR is super bare bones and does not implement everything. It was just really annoying to have all my data blow up whenever I would merge my chunks together. At the same time, I did not want to rewrite a compressed merge script for every use case.

The solution I landed on was to use a small config file to specify the most impactful options like compression, compression_level, set_dictionary_enabled and also column level encodings.

Putting this PR up in case you are interested in using it in the main branch. It would be cool to not have to tell people to install my fork haha.

Thanks

### CODE REVIEW

**Review by trisha — COMMENTED** [review]
happy open source friday!

**Comment by CalderWhite** [comment]
> happy open source friday!

LOLLL

File: src/commands/merge.rs [hunk 1]
```diff
@@ -16,12 +22,115 @@ pub struct MergeCommandArgs {
...
+        let encoding_mappings = build_encoding_mappings();
+        for (column_name, encoding_str) in column_encodings {
+            if !encoding_mappings.contains_key(encoding_str.as_str()) {
+                return Err(PQRSError::IllegalEncodingType());
+            }
+
+            let encoding = *encoding_mappings
+                .get(encoding_str.clone().as_str())
+                .unwrap();
+            props = props.set_column_encoding(ColumnPath::from(column_name), encoding)
```
Comment (manojkarthick) [inline src/commands/merge.rs:77]: can this be made idiomatic use pattern matching?

File: src/commands/merge.rs [hunk 2]
```diff
@@ -16,12 +22,115 @@ pub struct MergeCommandArgs {
...
+        }
+    }
+
+    if let Some(column_de) = merge_config.column_dictionary_enabled {
+        for (column_name, de) in column_de {
+            println!("{column_name}");
```
Comment (manojkarthick) [inline src/commands/merge.rs:83]: we probably don't want `println!()`s in the release

File: src/commands/merge.rs [hunk 3]
```diff
@@ -16,12 +22,115 @@ pub struct MergeCommandArgs {
...
+            props =
+                props.set_column_dictionary_enabled(ColumnPath::from(column_name), de);
+        }
+    }
+
+    if let Some(compression_algo) = merge_config.compression {
+        if compression_algo.to_lowercase() == "brotli" {
```
Comment (manojkarthick) [inline src/commands/merge.rs:90]: same for `compression_algo`, I think we can use pattern matching here?

File: src/main.rs [hunk 4]
```diff
@@ -6,6 +6,11 @@ mod commands;
 mod errors;
 mod utils;
 
+// use jemalloc for release builds
+extern crate jemallocator;
+#[global_allocator]
+static ALLOC: jemallocator::Jemalloc = jemallocator::Jemalloc;
+
```
Comment (manojkarthick) [inline src/main.rs:13]: i'm unfamiliar with `jemallocator`. can you give a tl;dr on these performance optimizations? what do they do?

**Review by manojkarthick — COMMENTED** [review]
Hey thanks for the PR! I left a few comments.

File: src/main.rs (hunk 4 above)
Comment (CalderWhite) [reply src/main.rs:13]: Is there a way to only use my first commit in the PR? The second commit is performance optimized stuff that only belongs on my fork.

TL;DR this can be removed for the main branch

File: src/commands/merge.rs (hunk 3 above)
Comment (CalderWhite) [reply src/commands/merge.rs:90]: Good callout!

File: src/commands/merge.rs [hunk 5]
```diff
@@ -16,12 +22,115 @@ pub struct MergeCommandArgs {
...
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct MergeConfig {
+    pub set_dictionary_enabled: Option<bool>,
+    /// The encodings for this are the just text values of the enum parquet::basic::Encoding
+    pub column_encodings: Option<HashMap<String, String>>,
+    pub column_dictionary_enabled: Option<HashMap<String, bool>>,
+    pub compression: Option<String>,
+    pub compression_level: Option<u32>,
+}
```
[the remark this answers is not in the document]
Comment (CalderWhite) [reply src/commands/merge.rs:39]: When using CLI opts your command blows up to be huge when you have a large number of columns. Using a JSON file is super convenient and easy to read :)

**Comment by CalderWhite** [comment]
Responded to all of your comments! Some of them can be completely removed from the PR, others require some modification and finally the config file is a matter of your opinion for the main repo.
